astropy/visualization/wcsaxes/coordinate_range.py

```diff
@@ -0,0 +1,127 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+from __future__ import print_function, division, absolute_import
+
+import warnings
+
+import numpy as np
+
+from ... import units as u
+
+from . import settings
+
+# Algorithm inspired by PGSBOX from WCSLIB by M. Calabretta
+
+
+def wrap_180(values):
+    values_new = values % 360.
+    with np.errstate(invalid='ignore'):
+        values_new[values_new > 180.] -= 360
+    return values_new
+
+
+def find_coordinate_range(transform, extent, coord_types, coord_units):
+    '''
+    Find the range of coordinates to use for ticks/grids
+
+    Parameters
+    ----------
+    pix2world : func
+        Function to transform pixel to world coordinates. Should take two
+        values (the pixel coordinates) and return two values (the world
+        coordinates).
+    extent : iterable
+        The range of the image viewport in pixel coordinates, given as [xmin,
+        xmax, ymin, ymax].
+    coord_types : list of str
+        Whether each coordinate is a ``'longitude'``, ``'latitude'``, or
+        ``'scalar'`` value.
+    coord_units : list of `astropy.units.Unit`
+        The units for each coordinate
+    '''
+
+    # Sample coordinates on a NX x NY grid.
+    NX = NY = settings.COORDINATE_RANGE_SAMPLES
+    x = np.linspace(extent[0], extent[1], NX + 1)
+    y = np.linspace(extent[2], extent[3], NY + 1)
+    xp, yp = np.meshgrid(x, y)
+    world = transform.transform(np.vstack([xp.ravel(), yp.ravel()]).transpose())
+
+    ranges = []
+
+    for coord_index, coord_type in enumerate(coord_types):
+
+        xw = world[:, coord_index].reshape(xp.shape)
+
+        if coord_type in ['longitude', 'latitude']:
+
+            unit = coord_units[coord_index]
+            xw = xw * unit.to(u.deg)
+
+            # Iron out coordinates along first row
+            wjump = xw[0, 1:] - xw[0, :-1]
+            with np.errstate(invalid='ignore'):
+                reset = np.abs(wjump) > 180.
+            if np.any(reset):
+                wjump = wjump + np.sign(wjump) * 180.
+                wjump = 360. * (wjump / 360.).astype(int)
+                xw[0, 1:][reset] -= wjump[reset]
+
+            # Now iron out coordinates along all columns, starting with first row.
+            wjump = xw[1:] - xw[:1]
+            with np.errstate(invalid='ignore'):
+                reset = np.abs(wjump) > 180.
+            if np.any(reset):
+                wjump = wjump + np.sign(wjump) * 180.
+                wjump = 360. * (wjump / 360.).astype(int)
+                xw[1:][reset] -= wjump[reset]
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", RuntimeWarning)
+            xw_min = np.nanmin(xw)
+            xw_max = np.nanmax(xw)
+
+        # Check if range is smaller when normalizing to the range 0 to 360
```

Comment:
In this part, there seems to be quite some overlap with functionality in `Angle` -- not really an issue for merging this, but it may be good to notice things like these in a separate issue.